checker/cost.go

```diff
@@ -599,3 +612,12 @@ func (c *coster) newAstNode(e *exprpb.Expr) *astNode {
 	}
 	return &astNode{path: path, t: c.getType(e), expr: e, derivedSize: derivedSize}
 }
+
+func isScalar(t *exprpb.Type) bool {
+	if kindOf(t) == kindPrimitive {
+		if t.GetPrimitive() != exprpb.Type_STRING && t.GetPrimitive() != exprpb.Type_BYTES {
```

Comment:
Should kind *exprpb.Type_WellKnown also be handled for the exprpb.Type_DURATION and exprpb.Type_TIMESTAMP cases somewhere in the return type checking?

Reply:
I also see *exprpb.Type_MessageType has "google.protobuf.Duration" and "google.protobuf.Timestamp" cases. I don't know if wellknown or message checks would be needed here (or both? or neither?).  Do we need tests for these types?

Reply:
Within the `exprpb.Type`, if any of the following fields are set, it's probably reasonable to consider the type to have variable size:

- `message_type`* - you might want to check to see if the message_type is a well-known `google.protobuf` type name, such as `google.protobuf.Int64Value`, `google.protobuf.Duration`, etc.
- `abstract_type` - custom type whose contents we cannot inspect without accessor methods.
- `well_known`* - only the `ANY` type should be consider an object.
- `list_type` 
- `map_type`

I think that's the bulk of the type set that can be considered variable-size.

Reply:
I've changed `isScalar` to now handle other types besides `kindPrimitive`, so #573 can be closed if this PR merges/I'm marking this as resolved. On a sidenote, I came across [ExprTypeToType](https://github.com/google/cel-go/blob/25bb4c6da1759f0104ce8ee935431d32b71977b6/cel/decls.go#L993) today, which could simplify `isScalar` to something like:
```go
func isScalar(t *Type) bool {
    celType, err := ExprTypeToType(t)
    if err != nil {
        return false
    }
    _, hasIndexerTrait := celType.(traits.Indexer)
    return !hasIndexerTrait
}
```
Unfortunately importing it results in a circular import, and I'm not sure I see an easy way out (make `isScalar` public and move it to `cel/decls`? Move `ExprTypeToType`?), but I thought I'd put it out there just in case.

Reply:
I'll end up moving the implementation of `ExprTypeToType` down into common, but it's part of a much larger refactor. For now, I'd stick to a deliberate test against the `exprpb.Type` oneof